chain/transaction.go

```diff
@@ -122,18 +122,18 @@ func (t *Transaction) Expiry() int64 { return t.Base.Timestamp }
 
 func (t *Transaction) MaxFee() uint64 { return t.Base.MaxFee }
 
-func (t *Transaction) StateKeys(sm StateManager) (set.Set[string], error) {
+func (t *Transaction) StateKeys(sm StateManager) (set.Set[state.Key], error) {
 	if t.stateKeys != nil {
 		return t.stateKeys, nil
 	}
 
 	// Verify the formatting of state keys passed by the controller
 	actionKeys := t.Action.StateKeys(t.Auth.Actor(), t.ID())
 	sponsorKeys := sm.SponsorStateKeys(t.Auth.Sponsor())
-	stateKeys := set.NewSet[string](len(actionKeys) + len(sponsorKeys))
-	for _, arr := range [][]string{actionKeys, sponsorKeys} {
+	stateKeys := set.NewSet[state.Key](len(actionKeys) + len(sponsorKeys))
+	for _, arr := range [][]state.Key{actionKeys, sponsorKeys} {
 		for _, k := range arr {
-			if !keys.Valid(k) {
+			if !keys.Valid(k.Name) {
 				return nil, ErrInvalidKeyValue
 			}
 			stateKeys.Add(k)
```

Comment:
How do you handle cases of conflicting keys? What happens if you insert a R/W key then later add a R key?

Reply:
With the map now, I tried to have it behave like a `set`. That is, a key that already exists in the map (i.e., say with RW) will still have RW. Therefore, adding a R key later will have no change. This is done using the `Add` function that I added in `state.go`. 

I also added tests in `tstate_test.go` to check this behavior too